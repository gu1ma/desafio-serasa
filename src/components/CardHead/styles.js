import styled from 'styled-components/native';
import LinearGradient from 'react-native-linear-gradient';

export const Container = styled(LinearGradient)`
  justify-content: center;
  align-items: center;
  padding-top: 20px;
  padding-right: 30px;
  padding-left: 30px;
  padding-bottom: 80px;
`;

export const UserPhoto = styled.Image`
  height: 140px;
  width: 140px;
  border-radius: 70px;
  padding: 5px;
`;

export const OutlineCircle = styled.View`
  margin-top: -140px;
  height: 140px;
  width: 140px;
  padding: 5px;

  border-bottom-width: 5px;
  border-right-width: 5px;
  border-left-width: 5px;
  border-top-width: 5px;

  border-top-left-radius: 70px;
  border-top-right-radius: 70px;
  border-bottom-left-radius: 70px;
  border-bottom-right-radius: 70px;

  border-top-color: ${props => props.colorBorder};
  border-left-color: ${props =>
    props.userLevel > 0 ? props.colorBorder : '#fff'};
  border-bottom-color: ${props =>
    props.userLevel > 1 ? props.colorBorder : '#fff'};
  border-right-color: ${props =>
    props.userLevel > 2 ? props.colorBorder : '#fff'};

  transform: rotate(-45deg);
`;

export const UserName = styled.Text`
  font-size: 26px;
  color: white;
  font-weight: bold;
  margin-top: 15px;
`;
