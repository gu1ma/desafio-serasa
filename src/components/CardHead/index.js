import React from 'react';
import ProgressCircle from 'react-native-progress-circle';
import {Container, UserPhoto, UserName} from './styles';

export default function CardHead({
  userData,
  initialColor,
  finalColor,
  borderColor,
}) {
  return (
    <Container colors={[initialColor, finalColor]}>
      <ProgressCircle
        borderWidth={5}
        percent={100 - userData.score}
        shadowColor={borderColor}
        radius={70}
        color="#fff">
        <UserPhoto source={userData.userPhoto} testID="userphoto" />
      </ProgressCircle>
      <UserName testID="username">{userData.userName}</UserName>
    </Container>
  );
}
