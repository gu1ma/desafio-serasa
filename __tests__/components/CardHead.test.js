import React from 'react';
import {render} from '@testing-library/react-native';

import CardHead from '~/components/CardHead';

describe('CardHead', () => {
  const userData = {
    userName: 'Rick Sanches',
    userPhoto: {
      uri:
        'https://i.pinimg.com/originals/e6/22/d2/e622d2fdccb6fdf31133dd7ab273892a.jpg',
    },
    score: 25,
    scoreDescription: 'Sua pontuação é baixa',
  };

  it('Should have the userimage Image component', () => {
    const {getByTestId} = render(<CardHead userData={userData} />);

    expect(getByTestId('userphoto')).toBeTruthy();
  });

  it('Should have the usename Text component', () => {
    const {getByTestId} = render(<CardHead userData={userData} />);

    expect(getByTestId('username')).toBeTruthy();
  });
});
