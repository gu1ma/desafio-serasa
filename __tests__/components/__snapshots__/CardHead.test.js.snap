// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHead Should render correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      undefined,
      undefined,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 80,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#999",
          "borderRadius": 70,
          "height": 140,
          "width": 140,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 140,
            "width": 70,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomRightRadius": 0,
              "borderTopRightRadius": 0,
              "left": 0,
              "position": "absolute",
              "top": 0,
            },
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 70,
              "height": 140,
              "overflow": "hidden",
              "transform": Array [
                Object {
                  "translateX": 35,
                },
                Object {
                  "rotate": "180deg",
                },
                Object {
                  "translateX": -35,
                },
              ],
              "width": 70,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 140,
            "width": 70,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomRightRadius": 0,
              "borderTopRightRadius": 0,
              "left": 0,
              "position": "absolute",
              "top": 0,
            },
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 70,
              "height": 140,
              "overflow": "hidden",
              "transform": Array [
                Object {
                  "translateX": 35,
                },
                Object {
                  "rotate": "270deg",
                },
                Object {
                  "translateX": -35,
                },
              ],
              "width": 70,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#e9e9ef",
            "borderRadius": 65,
            "height": 130,
            "width": 130,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://i.pinimg.com/originals/e6/22/d2/e622d2fdccb6fdf31133dd7ab273892a.jpg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 70,
              "height": 140,
              "paddingBottom": 5,
              "paddingLeft": 5,
              "paddingRight": 5,
              "paddingTop": 5,
              "width": 140,
            },
          ]
        }
      />
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 26,
          "fontWeight": "bold",
          "marginTop": 15,
        },
      ]
    }
  >
    Rick Sanches
  </Text>
</BVLinearGradient>
`;
